lisa/analysis/rta.py

```diff
@@ -198,6 +198,64 @@ def save_plot(self, figure, filepath=None, img_format=None):
         default_dir = dirnames.pop()
         return self._save_plot(figure, default_dir, filepath, img_format)
 
+    @memoized
+    def df_activations(self, task, start_time=None):
+        """
+        Get activation events
+
+        :param task: Name of the task that we want the performance dataframe of.
+        :type task: str
+
+        :param start_time: The first activation time in seconds
+        :type start_time: float
+
+        :returns: A :class:`pandas.DataFrame` with index representing the start
+        time of an activation and these columns:
+
+        * ``Run``: the running time of the activation
+        * ``End``: the end time of the activation
+        * ``Slack``: the slack of the activation
+        * ``WKPLatency``: the wakeup latency of the activation
+        * ``PerfIndex``: the performance index of the activation
+        """
+        activations_df = self.get_df(task, start_time)[[
+            'Run', 'Slack', 'WKPLatency', 'PerfIndex']].copy()
+        activations_df['End'] = activations_df.index + activations_df['Run'] / 1e6
+
+        # Reorder to keep "End" as the second column
+        return activations_df[['Run', 'End', 'Slack', 'WKPLatency', 'PerfIndex']]
+
+    def plot_activations(self, task, start_time=None, filepath=None, axis=None):
```

Comment:
Use `**kwargs` instead of `filepath=None, axis=None`

Reply:
I hate those variable args... you never know what a function can take as params.

We reference the documentation of:

   `lisa.analysis.base.AnalysisHelpers.do_plot`

which correctly lists only `filepath`, `axis` and other forwarded `kwargs` params.
But, if you look at the code:

  https://lisa-linux-integrated-system-analysis.readthedocs.io/en/master/_modules/lisa/analysis/base.html#TraceAnalysisBase

you see an additional `plotter` parameter and that's confusing.

Anyway, I'll update the code to what we use also for the other plot functions.

Reply:
> I hate those variable args... you never know what a function can take as params.

That's what the `.. seealso` is supposed to address, all analysis functions are relying on the same helper.

> you see an additional plotter parameter and that's confusing.

True.

That is unfortunately the only way to be able to add new `do_plot` arguments without having to edit the whole (growing) set of plot functions, so we don't really have the choice here. We can however improve the documentation side. #917 introduces a decorator, which may be used to add extra explicit parameters to the docstring like we do with `RTATestBundle.check_noisy_tasks`. We would only add the relevant arguments from `do_plot` so it is not polluted by `plotter`.